containers/src/Kokkos_DynamicView.hpp

```diff
@@ -170,13 +170,20 @@ struct ChunkedArrayManager {
           m_chunk_max(arg_chunk_max),
           m_chunk_size(arg_chunk_size) {}
 
+    // Concurrently destroy array of chunk pointers.
+    KOKKOS_INLINE_FUNCTION void operator()(unsigned i) const {
+      Space().deallocate(m_label.c_str(), m_chunks[i],
+                         sizeof(value_type) * m_chunk_size);
+    }
+
     void execute() {
       // Destroy the array of chunk pointers.
-      // Two entries beyond the max chunks are allocation counters.
-      for (unsigned i = 0; i < m_chunk_max; i++) {
-        Space().deallocate(m_label.c_str(), m_chunks[i],
-                           sizeof(value_type) * m_chunk_size);
-      }
+      using ExecSpace = typename MemorySpace::execution_space;
+      using Range = Kokkos::RangePolicy<unsigned, ExecSpace>;
+      Kokkos::Impl::ParallelFor<Destroy<Space>, Range> closure(*this, Range(0, m_chunk_max));
+      closure.execute();
+      ExecSpace().fence();
```

Comment:
Is there a reason not to use a regular `Kokkos::parallel_for` here? Also, I'm a little concerned about deallocating data in parallel since deallocations should fence. Why isn't that a problem here?

Reply:
yeah this isn't valid you are technically not allowed to deallocated inside Kokkos parallel operations.